libraries/pbrlib/genglsl/mx_conductor_brdf.glsl

```diff
@@ -38,30 +32,24 @@ void mx_conductor_brdf_reflection(vec3 L, vec3 V, float weight, vec3 reflectivit
     result = D * F * G * comp * weight / (4 * NdotV);
 }
 
-void mx_conductor_brdf_indirect(vec3 V, float weight, vec3 reflectivity, vec3 edge_color, vec2 roughness, vec3 N, vec3 X, int distribution, out vec3 result)
+void mx_conductor_brdf_indirect(vec3 V, float weight, vec3 reflectivity, vec3 edge_color, vec2 roughness, vec3 N, vec3 X, int distribution, out BSDF result)
 {
     if (weight < M_FLOAT_EPS)
-    {
-        result = vec3(0.0);
-        return;
-    }
-
-    float NdotV = dot(N,V);
-    if (NdotV <= 0.0)
     {
         result = BSDF(0.0);
         return;
     }
 
+    float NdotV = clamp(dot(N, V), M_FLOAT_EPS, 1.0);
+
     vec3 ior_n, ior_k;
     mx_artistic_to_complex_ior(reflectivity, edge_color, ior_n, ior_k);
 
     float avgRoughness = mx_average_roughness(roughness);
-
-    vec3 Li = mx_environment_radiance(N, V, X, roughness, vec3(1.0), vec3(1.0), distribution);
-    vec3 F = mx_fresnel_conductor(dot(N, V), ior_n, ior_k);
-
+    vec3 F = mx_fresnel_conductor(NdotV, ior_n, ior_k);
     vec3 comp = mx_ggx_energy_compensation(NdotV, avgRoughness, F);
 
-    result = Li * F * comp * weight;
+    vec3 Li = mx_environment_radiance(N, V, X, roughness, vec3(1.0), vec3(1.0), ior_n, ior_k, distribution, 1);
```

Comment:
For clarity maybe we could define "enum" constants for the `fresnelMode` values.

Reply:
That's a good idea, and I was thinking of making a change along those lines in an upcoming revision.